pkg/kubelet/cm/cpumanager/policy_static.go

```diff
@@ -354,6 +354,74 @@ func (p *staticPolicy) GetTopologyHints(s state.State, pod *v1.Pod, container *v
 	}
 }
 
+func (p *staticPolicy) getPodRequestedCPU(pod *v1.Pod) int {
+	// The maximum of requested CPUs by init containers.
+	requestedByInitContainers := 0
+	for _, container := range pod.Spec.InitContainers {
+		if _, ok := container.Resources.Requests[v1.ResourceCPU]; !ok {
+			continue
+		}
+		requestedCPU := p.guaranteedCPUs(pod, &container)
+		if requestedCPU > requestedByInitContainers {
+			requestedByInitContainers = requestedCPU
+		}
+	}
+	// The sum of requested CPUs by app containers.
+	requestedByAppContainers := 0
+	for _, container := range pod.Spec.Containers {
+		if _, ok := container.Resources.Requests[v1.ResourceCPU]; !ok {
+			continue
+		}
+		requestedByAppContainers += p.guaranteedCPUs(pod, &container)
+	}
+
+	requestedByPod := requestedByAppContainers
+	if requestedByInitContainers > requestedByAppContainers {
+		requestedByPod = requestedByInitContainers
+	}
+	klog.Errorf("[cpumanager] requestedByInits : %d, requestedByUsers : %d, requestedByPod : %d", requestedByInitContainers, requestedByAppContainers, requestedByPod)
+	return requestedByPod
+}
+
+func (p *staticPolicy) GetPodLevelTopologyHints(s state.State, pod *v1.Pod) map[string][]topologymanager.TopologyHint {
+	// Get a count of how many guaranteed CPUs have been requested by Pod.
+	requestedByPod := p.getPodRequestedCPU(pod)
+
+	if requestedByPod == 0 {
+		return nil
+	}
+
+	assignedCPUs := cpuset.NewCPUSet()
+	for _, container := range append(pod.Spec.InitContainers, pod.Spec.Containers...) {
+		requested := p.guaranteedCPUs(pod, &container)
+		if allocated, exists := s.GetCPUSet(string(pod.UID), container.Name); exists {
+			if allocated.Size() != requested {
+				klog.Errorf("[cpumanager] CPUs already allocated to (pod %v, container %v) with different number than request: requested: %d, allocated: %d", string(pod.UID), container.Name, requested, allocated.Size())
+				return map[string][]topologymanager.TopologyHint{
+					string(v1.ResourceCPU): {},
+				}
+			}
+			assignedCPUs = assignedCPUs.Union(allocated)
+		}
+	}
+	if assignedCPUs.Size() == requestedByPod {
+		klog.Infof("[cpumanager] Regenerating TopologyHints for CPUs already allocated to pod %v", string(pod.UID))
+		return map[string][]topologymanager.TopologyHint{
+			string(v1.ResourceCPU): p.generateCPUTopologyHints(assignedCPUs, requestedByPod),
+		}
+	}
+
+	// Get a list of available CPUs.
+	available := p.assignableCPUs(s)
+
+	// Generate hints.
+	cpuHints := p.generateCPUTopologyHints(available, requestedByPod)
+	klog.Infof("[cpumanager] TopologyHints generated for pod '%v' : %v", pod.Name, cpuHints)
```

Comment:
`pod.Name` would be great to include also namespace. Or use `UID` as above.

Reply:
Right, corrected it in provided places.